iaso/api/storage.py

```diff
@@ -0,0 +1,334 @@
+    class Meta:
+        model = OrgUnit
+        fields = [
+            "id",
+            "name",
+        ]
+
+
+# This is actually unused (by POST)
+class StorageLogSerializer(serializers.ModelSerializer):
+    storage_id = serializers.CharField(source="device.customer_chosen_id")
+    storage_type = serializers.CharField(source="device.type")
+    entity = EntityNestedSerializer(read_only=True)
+    org_unit = OrgUnitNestedSerializer(read_only=True)
+    performed_at = TimestampField(read_only=True)
+
+    class Meta:
+        model = StorageLogEntry
+        fields = [
+            "id",
+            "storage_id",
+            "storage_type",
+            "operation_type",
+            "instances",
+            "org_unit",
+            "entity",
+            "performed_at",
+        ]
+
+    def create(self, validated_data):
+        pass
+
+
+class StorageStatusSerializer(serializers.Serializer):
+    status = serializers.ChoiceField(choices=StorageDevice.STATUS_CHOICES)
+    reason = serializers.ChoiceField(
+        source="status_reason", required=False, choices=StorageDevice.STATUS_REASON_CHOICES
+    )
+    # TODO: where should this data come from
+    # updated_at = serializers.DateTimeField()
+    # TODO: Comment field is not in the spec, but I guess it's useful to implement, no?
+    comment = serializers.CharField(source="status_comment", required=False, allow_blank=True)
+
+    def validate(self, data):
+        """
+        Ensure that a reason is set if changed to a non-ok status
+        """
+        missing_status_reason = True
+        if "status_reason" in data and data["status_reason"] != "":
+            missing_status_reason = False
+
+        if data["status"] != StorageDevice.OK and missing_status_reason:
+            raise serializers.ValidationError("A reason must be provided when changing the status to a non-ok status")
+        return data
+
+
+class StorageSerializer(serializers.ModelSerializer):
+    storage_id = serializers.CharField(source="customer_chosen_id")
+    storage_type = serializers.CharField(source="type")
+    status = StorageStatusSerializer(source="*")
+    entity = EntityNestedSerializer(read_only=True)
+    org_unit = OrgUnitNestedSerializer(read_only=True)
+    created_at = TimestampField(read_only=True)
+    updated_at = TimestampField(read_only=True)
+
+    class Meta:
+        model = StorageDevice
+        fields: Tuple[str, ...] = (
+            "updated_at",
+            "created_at",
+            "storage_id",
+            "storage_type",
+            "status",
+            "org_unit",
+            "entity",
+        )
+
+
+class StorageSerializerWithLogs(StorageSerializer):
+    """Like StorageSerializer, but also includes the log entries"""
+
+    logs = StorageLogSerializer(many=True, source="log_entries")
+
+    class Meta(StorageSerializer.Meta):
+        fields = StorageSerializer.Meta.fields + ("logs",)
+
+
+class StorageViewSet(ListModelMixin, viewsets.GenericViewSet):
```

Comment:
if you use iaso common.ModelViewSet so you don't need to reimplement the list method as it handle pagination in the expected way